src/Symfony/Component/AccessToken/Bridge/OAuth/ClientCredentials.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AccessToken\Bridge\OAuth;
+
+/**
+ * OAuth2 "client_credentials" grant type.
+ *
+ * @see https://www.oauth.com/oauth2-servers/access-tokens/client-credentials/
+ *
+ * @author Pierre Rineau <pierre.rineau@processus.org>
+ */
+class ClientCredentials extends AbstractOAuthCredentials
+{
+    use WithScopeTrait;
+
+    /**
+     * @param string                    $clientId     client ID
+     * @param string                    $clientSecret client secret
+     * @param string|null               $tenant       tenant name or identifier
+     * @param string|array<string>|null $scope        requested scopes, either as a whitespace separated list or an array of strings
+     * @param string|null               $endpoint     authorization endpoint URL, for generic usage you must provide one
+     */
+    public function __construct(
+        #[\SensitiveParameter] private readonly string $clientId,
+        #[\SensitiveParameter] private readonly string $clientSecret,
+        #[\SensitiveParameter] ?string $tenant = null,
+        string|array|null $scope = null,
+        ?string $endpoint = null,
+    ) {
+        parent::__construct(
+            tenant: $tenant,
+            endpoint: $endpoint,
+        );
```

Comment:
oneliner and no named args please

Reply:
Are named args against the Symfony convention ? Fabbot.io doesn't complain about it. But in this particular case I'm OK with reverting to match the parent constructor because it's a simple one.